src/libipc/client_api/client_api.c

```diff
@@ -155,14 +155,23 @@ int ipc_recv(char src[NAME_LEN], char * buffer, size_t buffer_len)
     }
   }
 
+  // Create placeholder for source name and message 
+  char name[NAME_LEN];
+  char msg_final[MAX_MSG_LEN];
+  size_t msg_final_len = 0;
+
+  // Separate message from source name 
+  strncpy(name, msg, NAME_LEN);
+  for(int x = NAME_LEN + 1; x < bytes_read; x++, msg_final_len++) msg_final[x-(NAME_LEN+1)] = msg[x];
```

Comment:
not a huge fan of this formatting

Reply:
yeah that's a pretty clumsy implementation. what about this?
```suggestion
  int msg_final_len = 0;
  for(int x = 0; x < bytes_read; x++) {
    if(x < NAME_LEN) { // extract name
      name[x] = msg[x];
    } else if(x > NAME_LEN) { // extract msg_final
      msg_final[msg_final_len] = msg[x];
      msg_final_len++;
    }
```

Reply:
I don't mind the `strncpy` part, thats perfectly fine. I was moreso pointing out the one line for loop. 
```suggestion
  for(int x = NAME_LEN + 1; x < bytes_read; x++, msg_final_len++) 
     msg_final[x-(NAME_LEN+1)] = msg[x];
```

Reply:
ohhh gotcha, yeah that's no worries! I thought the black magic `msg_final[x-(NAME_LEN+1)]` stuff was the problem haha.

-- newline approved --

Reply:
Up to you though!